### DISCUSSION TITLE
User Extras

### DISCUSSION THREAD

**Discussion by andresatierf** [LazyVim/LazyVim#2734] [Q&A] [answered]
I can't for the life of me figure out how to use the user extras.

I have created an `extras` directory inside the `plugins` directory. I added a spec for a plugin. Nothing shows up in the `Lazy Extras` window (besides de default extras).

Can someone give me a hand with this?

*Screenshots*
![image](https://github.com/LazyVim/LazyVim/assets/29847057/8ec4fccc-df3c-4750-9c9f-54b0d32c8f65)
![image](https://github.com/LazyVim/LazyVim/assets/29847057/34f94f4b-ee28-4cc2-a842-c066ca9d944c)

**Comment by jyuan0** [comment]
`LazyExtras`

https://www.lazyvim.org/extras

**Reply by andresatierf** [reply]
That is not helpful at all. That only opens the lazy extras window, which you can see from the screenshots that I already have open.

**Reply by jyuan0** [reply]
Sorry, I didn't realize the screenshots were of the Extras panel.

At the top of the panel, it says `Enable/disable extras with the <x> key`.

**Reply by andresatierf** [reply]
I know. My problem is that my `UserExtras` do not show up on that list

**Reply by jyuan0** [reply]
Are `UserExtras` plugins you've added? I'm not sure the Extras panel is supposed to show those.

**Reply by andresatierf** [reply]
I think it should since [this](https://github.com/LazyVim/LazyVim/issues/1681) was merged

**Reply by andresatierf** [reply]
maybe @folke can shine some light on this?

**Reply by jyuan0** [reply]
> I think it should since https://github.com/LazyVim/LazyVim/issues/1681 was merged

Oh sorry for the noise. I had no idea that was possible.

**Reply by andresatierf** [reply]
I only found out yesterday myself and decided that I wanted to try it out

**Reply by dpetka2001** [reply]
@andresatierf I have tried exactly your config and it shows up in the LazyExtras for me as a `User` Extra.

![2024-03-15_19-35](https://github.com/LazyVim/LazyVim/assets/12776461/5396d3f3-aa77-4bed-88ed-c17a9048348f)

![2024-03-15_19-36](https://github.com/LazyVim/LazyVim/assets/12776461/535a2044-8452-45b8-9933-a9d7dcf10bf2)

**Reply by jyuan0** [reply]
I haven't tried enabling/disabling, but it I think the plugin spec should be in a directory in `extras` - e.g. `lua/plugins/extras/ui/icon-picker.lua` (it's listed in Extras panel).

Update: no, I'm wrong, it should show up even in `extras` like @dpetka2001 said.

**Reply by dpetka2001** [reply]
No, it doesn't seem to be necessary. I put directly in the `plugins/extras/` directory myself and it correctly showed up in the LazyExtras. You can just try it for yourself and check if it shows up or not.

**Reply by jyuan0** [reply]
Yes, you're right, it should show up even in `plugins/extras`. Sorry for the noise again.

**Reply by dpetka2001** [reply]
I don't consider it noise from your part, so no need to worry :)

**Reply by andresatierf** [reply, marked as answer]
I figured out what the problem was. I use GNU stow to manage my dotfiles so the file in the extras folder is actually a symbolic link, as seen here:
![image](https://github.com/LazyVim/LazyVim/assets/29847057/c3651c27-18f4-402e-a3ac-a069d41508be)

When I copy the file directly into the directory, then it shows up in the `LazyExtras` screen.
